src/functions.h

```diff
@@ -2,172 +2,166 @@
-  cli();
-#endif
-	const unsigned long Time = micros();
-	const unsigned long  duration = Time - lastTime;
-	lastTime = Time;
-	if (duration >= pulseMin) {//kleinste zulaessige Pulslaenge
-		int sDuration;
-		if (duration < maxPulse) {//groesste zulaessige Pulslaenge, max = 32000
-			sDuration = int(duration); //das wirft bereits hier unnoetige Nullen raus und vergroessert den Wertebereich
-		}
-		else {
-			sDuration = maxPulse; // Maximalwert set to maxPulse defined in lib.
-		}
-		if (isHigh(PIN_RECEIVE)) { // Wenn jetzt high ist, dann muss vorher low gewesen sein, und dafuer gilt die gemessene Dauer.
-			sDuration = -sDuration;
-		}
-		FiFo.enqueue(sDuration);
-	} // else => trash
-#ifdef ARDUINO_MAPLEMINI_F103CB
-  interrupts();
-#else
-  sei();
-#endif
+  #ifdef ARDUINO_MAPLEMINI_F103CB
+    noInterrupts();
+  #else
+    cli();
+  #endif
+  const unsigned long Time = micros();
+  const unsigned long  duration = Time - lastTime;
+  lastTime = Time;
+  if (duration >= pulseMin) {//kleinste zulaessige Pulslaenge
+    int sDuration;
+    if (duration < maxPulse) {//groesste zulaessige Pulslaenge, max = 32000
+      sDuration = int(duration); //das wirft bereits hier unnoetige Nullen raus und vergroessert den Wertebereich
+    }
+    else {
+      sDuration = maxPulse; // Maximalwert set to maxPulse defined in lib.
+    }
+    if (isHigh(PIN_RECEIVE)) { // Wenn jetzt high ist, dann muss vorher low gewesen sein, und dafuer gilt die gemessene Dauer.
+      sDuration = -sDuration;
+    }
+    FiFo.enqueue(sDuration);
+  } // else => trash
+  #ifdef ARDUINO_MAPLEMINI_F103CB
+    interrupts();
+  #else
+    sei();
+  #endif
 }
 
 
 void enableReceive() {
-	attachInterrupt(digitalPinToInterrupt(PIN_RECEIVE), handleInterrupt, CHANGE);
-#ifdef CMP_CC1101
-	if (hasCC1101) cc1101::setReceiveMode();
-#endif
+  // ToDo MR if(cc1101::ccmode == 0) ???
+  attachInterrupt(digitalPinToInterrupt(PIN_RECEIVE), handleInterrupt, CHANGE);
+  #ifdef CMP_CC1101
+    if (hasCC1101) cc1101::setReceiveMode();
+  #endif
 }
 
-void disableReceive() {
-	detachInterrupt(digitalPinToInterrupt(PIN_RECEIVE));
 
-#ifdef CMP_CC1101
-	if (hasCC1101) cc1101::setIdleMode();
-#endif
-	FiFo.flush();
+void disableReceive() {
+  // ToDo MR if(cc1101::ccmode == 0) ???
+  detachInterrupt(digitalPinToInterrupt(PIN_RECEIVE));
 
+  #ifdef CMP_CC1101
+    if (hasCC1101) cc1101::setIdleMode();
+  #endif
+  FiFo.flush();
 }
 
-//================================= EEProm commands ======================================
 
-void storeFunctions(const int8_t ms, int8_t mu, int8_t mc, int8_t red)
-{
-	mu = mu << 1;
-	mc = mc << 2;
-	red = red << 3;
-
-	int8_t dat = ms | mu | mc | red;
-	EEPROM.write(addr_features, dat);
-	#if defined(ESP8266) || defined(ESP32)
-	EEPROM.commit();
-	#endif
+//================================= EEProm commands ======================================
+void storeFunctions(const int8_t ms, int8_t mu, int8_t mc, int8_t red) {
+  mu = mu << 1;
+  mc = mc << 2;
+  red = red << 3;
```

Comment:
Was ist denn red für eine Option?

Reply:
@sidey79, das ist aber nun wieder eine Stelle die auch schon im dev-r34 drin war wo du @elektron-bbs und mir den Stein "zuwerfen" möchtest mit einer Frage die nicht dem PR geschuldet. *grins*

https://github.com/RFD-FHEM/SIGNALDuino/blob/74fa9482db391db28d532693b42da8c63b6d369f/src/functions.h#L88

>Was ist denn red für eine Option?

Ohne den Code nun noch einmal angesehen zu haben, vermute ich die Option steht für die Reduction.

"analog" -> 
`config: MS=1;MU=1;MC=1;Mred=1` oder
`storeFunctions(musterDec.MSenabled, musterDec.MUenabled, musterDec.MCenabled, musterDec.MredEnabled);`

Reply:
Das verstehe ich jetzt nicht. Die Zeile wird hier hinzugefügt. Was soll das denn am Ende bewirken, die Reduktion im flash zu speichern? Soll die Änderung damit persistent gemacht werden (können) ?

Reply:
Wenn ich es richtig im Kopf habe, so kam die void.... Sache wo du gerade etwas nicht verstehst, nicht von mir herein. Aus diesem Grund kann ich dir auch nicht sagen was du nicht verstehst.

Reply:
äh, wo kamm den der parameter red her?

Reply:
Der Parameter kommt z.B. aus functions.h:
```
    storeFunctions(1, 1, 1, 1);    // Init EEPROM with all flags enabled
```
oder aus commands.h:
```
	storeFunctions(musterDec.MSenabled, musterDec.MUenabled, musterDec.MCenabled, musterDec.MredEnabled);
```

Ansonsten ist diese "void storeFunctions(const int8_t ms, int8_t mu, int8_t mc, int8_t red)" mindestens seit der Version "SIGNALDuino-dev-r33_2019-02-05" unverändert, bis auf eine Verschiebung der geschweiften Klammer vom Zeilenanfang auf das Zeilenende der vorhergehenden Zeile und Änderung der Einzüge von Tabs in Spaces.

Reply:
Okay, dann bin ich dem diff wohl auf den Leim gegeangen.
Dunkel erinnere ich mich, dass es schon mal Diskussioen dazu gab. Dass wir das aber in die firmware aufgenommen hatten war mir dann doch nicht bewusst.

Mir fällt da ja nichts ein, wieso jemand daueraft die Komprimierung deaktivieren sollte. :)

Reply:
Naja, ich hab sie z.B. bei den SIGNAL-ESP abgeschaltet, weil sie vermutlich da nicht notwendig ist.

Reply:
Kann man machen, hängt vielleicht vom genauen Einsatz ab.
Die Komprimierung sorgt dafür, dass der uC weniger zu senden hat und wieder schneller in das Hauptprogramm zurück kommt. Ist dir denn irgendwas negatives dabei aufgefallen oder wieso hast Du es deaktiviert?

Reply:
Ist eigentlich mehr ein Bauchgefühl, das die Übertragung mit 54 Mbit anstatt 57 kbit eigentlich keine Bremse mehr darstellen sollte und Controller und FHEM können sich das kodieren/dekodieren sparen.